.github/workflows/acceptance-tests.yml

```diff
@@ -32,42 +32,29 @@ jobs:
           AIVEN_TOKEN: ${{ secrets.AIVEN_TOKEN }}
           AIVEN_PROJECT_NAME_PREFIX: ${{ secrets.AIVEN_PROJECT_NAME_PREFIX }}
 
-  acceptance_tests:
-    needs: setup_aiven_project_suffix
+  find_tests:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.find_tests.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - id: find_tests
+        run: |
+          echo "matrix=$(find ./internal -type f -name '*_test.go' -exec dirname {} \; | sort -u | jq -cnR '[inputs | select(length>0)]')" >> $GITHUB_OUTPUT
+
+  test:
+    needs: [setup_aiven_project_suffix, find_tests]
     runs-on: ubuntu-latest
     env:
       ACC_TEST_PARALLELISM: 5
     strategy:
       max-parallel: 5
-      fail-fast: false
+      fail-fast: true
       matrix:
-        pkg: [
-          account,
-          alloydbomni,
-          cassandra,
-          clickhouse,
-          connectionpool,
-          dragonfly,
-          flink,
-          grafana,
-          influxdb,
-          kafka,
-          kafkaschema,
-          kafkatopic,
-          m3db,
-          mysql,
-          opensearch,
-          organization,
-          pg,
-          project,
-          redis,
-          servicecomponent,
-          serviceintegration,
-          staticip,
-          thanos,
-          valkey,
-          vpc,
-        ]
+        pkg_path: ${{ fromJson(needs.find_tests.outputs.matrix) }}
```

Comment:
nit: would be nice to have a package name in the matrix name instead of `test ./internal/...`